From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kirill Yatsenko <kiriyatsenko@gmail.com>
Date: Thu, 03 Apr 2026 14:00:00 +0200
Subject: dts: Add device tree for decktrix custom PCB

Decktrix v1 custom PCB based on STM32MP157F.
DSI disabled to avoid conflict with kernel Jadard driver.
PMIC LDO1 at 2.8V for display AVDD, LDO6 at 1.8V for display IOVCC.

Signed-off-by: Kirill Yatsenko <kiriyatsenko@gmail.com>
---
 arch/arm/dts/Makefile                  |  1 +
 arch/arm/dts/decktrix-v1.dts           | 51 +++++++++++++++++++++++++
 arch/arm/dts/decktrix-v1-u-boot.dtsi   |  6 +++
 3 files changed, 58 insertions(+)
 create mode 100644 arch/arm/dts/decktrix-v1.dts
 create mode 100644 arch/arm/dts/decktrix-v1-u-boot.dtsi

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 267b0179..78206161 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -1090,6 +1090,7 @@ dtb-$(CONFIG_STM32MP15X) += \
 	stm32mp157a-microgea-stm32mp1-microdev2.0-of7.dtb \
 	stm32mp157c-dk2.dtb \
 	stm32mp157c-dk2-scmi.dtb \
+	decktrix-v1.dtb \
 	stm32mp157c-ed1.dtb \
 	stm32mp157c-ed1-scmi.dtb \
 	stm32mp157c-ev1.dtb \
diff --git a/arch/arm/dts/decktrix-v1.dts b/arch/arm/dts/decktrix-v1.dts
new file mode 100644
index 00000000..c1ca7a1b
--- /dev/null
+++ b/arch/arm/dts/decktrix-v1.dts
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) Decktrix Lab
+ * Author: Kirill Yatsenko <kiriyatsenko@gmail.com>
+ *
+ * Decktrix v1 custom PCB based on STM32MP157F.
+ * DSI disabled to avoid conflict with kernel Jadard driver.
+ */
+
+/dts-v1/;
+
+#include "stm32mp157.dtsi"
+#include "stm32mp15xc.dtsi"
+#include "stm32mp15-pinctrl.dtsi"
+#include "stm32mp15xxac-pinctrl.dtsi"
+#include "stm32mp15xx-dkx.dtsi"
+
+/ {
+	model = "Decktrix v1";
+	compatible = "decktrix,v1", "st,stm32mp157";
+
+	aliases {
+		ethernet0 = &ethernet0;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+};
+
+&cryp1 {
+	status = "okay";
+};
+
+&dsi {
+	status = "disabled";
+};
+
+/* PMIC LDO1: 2.8V for display AVDD (not 1.8V for audio) */
+&v1v8_audio {
+	regulator-name = "ldo1";
+	regulator-min-microvolt = <2800000>;
+	regulator-max-microvolt = <2800000>;
+};
+
+/* PMIC LDO6: 1.8V for display IOVCC (not 1.2V for HDMI) */
+&v1v2_hdmi {
+	regulator-name = "ldo6";
+	regulator-min-microvolt = <1800000>;
+	regulator-max-microvolt = <1800000>;
+};
diff --git a/arch/arm/dts/decktrix-v1-u-boot.dtsi b/arch/arm/dts/decktrix-v1-u-boot.dtsi
new file mode 100644
index 00000000..384f306c
--- /dev/null
+++ b/arch/arm/dts/decktrix-v1-u-boot.dtsi
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) Decktrix Lab
+ */
+
+#include "stm32mp157a-dk1-u-boot.dtsi"
--
2.43.0

